### PR TITLE
Display connection error if front end cannot connect to API server #11

### PR DESCRIPTION
**Pull request by D-Lite**
https://user-images.githubusercontent.com/44473671/230721270-2b1eae44-d5ba-4d51-91f4-fa47ef75e8bf.mp4

Display connection error if front end cannot connect to API server:
 - Created a universal page guard that is used at the route level for each page highly dependent on the BE
 - Improvement on the loader ensuring health check is complete before any additional information.

### CODE REVIEW

**Review by ciminelli — COMMENTED** [review]
Looking great! One small comment then I think we should be good to go :D

File: frontend/src/util/HealthCheck.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  if (loading) {
+    return (
+      <div
+        className="container-fluid vw-100 vh-100 align-items-center d-flex justify-content-center"
+        style={{
+          background: "rgba(0,0,0,0.4)",
+          // opacity: 0.4,
+        }}
+      >
+        <LoadingOverlay active={loading} spinner={true} text="Loading..." />
```
[the remark this answers is not in the document]
Comment (D-Lite) [reply frontend/src/util/HealthCheck.js:52]: I made the wrong upload, thanks for pointing that out, I will replace it soon and we can proceed!

**Comment by ciminelli** [comment]
@D-Lite one thing I noticed is that the error page exists on http://localhost:3000/welcome

So after the connection issue is resolved the user is taken back there. Can you look into making it so the error is displayed on the same route the user is on?

File: frontend/src/App.js [hunk 2]
```diff
@@ -47,28 +50,56 @@ function App() {
...
       >
         <Switch>
-          <Route exact path="/" component={LayoutActionTable}/>
-          <Route exact path="/audit" component={LayoutAuditTable}/>
+          <Route exact path="/">
+            <CheckBEHealth>
+              <LayoutActionTable />
+            </CheckBEHealth>
+          </Route>
+          <Route exact path="/">
```
Comment (ciminelli) [inline frontend/src/App.js:58]: It looks like `/audit` was removed here if you can add it back!

**Review by ciminelli — CHANGES_REQUESTED** [review]
Looking good! Just one small fix and we can merge.

**Comment by D-Lite** [comment]
Thanks for that!
Fixed

**Review by ciminelli — APPROVED** [review]
looksfine

**Pull request merged**